lte/gateway/c/core/oai/test/sgw_s8_task/sgw_csr.cpp

```diff
@@ -15,59 +15,47 @@
 
 using ::testing::Test;
 
-/* TC validates creation of ue context, pdn context and bearer context
- * on reception of Create Session Req
- */
-TEST_F(SgwS8Config, create_context_on_cs_req) {
-  sgw_state_init(false, config);
-  sgw_state_t* sgw_state = get_sgw_state(false);
-
-  spgw_ue_context_t* ue_context_p = NULL;
-  mme_sgw_tunnel_t sgw_s11_tunnel = {0};
-  sgw_s11_tunnel.local_teid       = sgw_s8_generate_new_cp_teid();
-
-  // validates creation of UE context on reception of Create Session Req
+// TC validates successful handling of create session request
+TEST_F(SgwS8ConfigAndCreateMock, create_context_on_cs_req_success) {
+  sgw_state_t* sgw_state                        = get_sgw_state(false);
+  itti_s11_create_session_request_t session_req = {0};
+  fill_itti_csreq(&session_req, default_eps_bearer_id);
+  memcpy(session_req.apn, "internet", sizeof("internet"));
   EXPECT_EQ(
-      sgw_update_teid_in_ue_context(
-          sgw_state, imsi64, sgw_s11_tunnel.local_teid),
+      sgw_s8_handle_s11_create_session_request(sgw_state, &session_req, imsi64),
       RETURNok);
-  EXPECT_EQ(
-      hashtable_ts_get(
-          sgw_state->imsi_ue_context_htbl, (const hash_key_t) imsi64,
-          reinterpret_cast<void**>(&ue_context_p)),
-      HASH_TABLE_OK);
+  sgw_state_exit();
+  std::this_thread::sleep_for(
+      std::chrono::milliseconds(END_OF_TESTCASE_SLEEP_MS));
+}
```

Comment:
sleeps are never good. Why do you need it here?

Reply:
Adding some time to clear the contexts before exiting from test case;  Currently it's 500 ms,

Reply:
is this needed in other test?

Reply:
During execution of "make test_oai" rarely I saw some memory leaks inconsistently in spite of clearing the memory. Then I looked at the other modules; where sleep was introduced at the end of each test case, So I thought of adding sleep for all test cases. 
Let me reduced the duration to 250 ms and check the impacts

Reply:
After test case is completed, it showed up memory leak. UE context and sgw_eps_bearer context are not released.

Reply:
check out the recently merged PR https://github.com/magma/magma/pull/10721/files. It probably is due to TearDown implementation.

Reply:
Let me know if you do the same changes whether the issue still persists or not.